conn_test.go

```diff
@@ -48,21 +48,18 @@ func TestNewConn(t *testing.T) {
 	ls.buildup(func(ls *localServer, ln net.Listener) {
 		conn, err := ls.Accept()
 		st.Assert(t, err, nil)
-		sc := newConn(conn)
 		// Respond with greeting
-		err = sc.writeDataUnit([]byte(testXMLGreeting))
+		err = writeDataUnit(conn, []byte(testXMLGreeting))
 		st.Assert(t, err, nil)
-		var res Response
 		// Read logout message
-		err = sc.readResponse(&res)
+		_, err = parseDataUnit(conn)
```

Comment:
The test misses the `LimitedReader{N: 0}` problem because we're not passing a `LimitedReader` here.

Reply:
The code never passes the limited reader to `parseDataUnit`, it passes the `net.Conn`.